src/net.cpp

```diff
@@ -2516,10 +2516,15 @@ void RelayTransaction(const CTransaction &tx, const CDataStream &ss)
     {
         if (!pnode->fRelayTxes)
             continue;
+
         LOCK(pnode->cs_filter);
-        if (pnode->pfilter)
+        // If the bloom filter is not empty then a peer must have sent us a filter
+        // and we can assume this node is an SPV node.
+        if (pnode->pfilter && !pnode->pfilter->IsEmpty())
```

Comment:
To skip peers with a default filter this needs to check `!IsFull()`.  Checking `!IsEmpty()` is a slight optimization for peers with a non-default filter.

Reply:
I don't see what the difference is...  a default filter is initialized to being both full and empty, so if I check !full or !empty it's the same result ?  (it seems a little odd to me that we even initialize to full at the start, it's not very intuitive but I don't know the history around that.)

Reply:
Ok this is unique to BU.  Both bitcoinxt/master and bitcoin/master initialize to isEmpty(false).  So we'll just have an implementation difference.